src/aria/popups/PopupNavigationManager.js

```diff
@@ -433,5 +642,236 @@ function ViewportNavigationInterceptor() {
+}
+exports.ElementNavigationInterceptor = ElementNavigationInterceptor;
+
+
+
+////////////////////////////////////////////////////////////////////////////////
+// Hiding manager
+////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Manages the hiding/showing of elements through the WAI-ARIA attribute "aria-hidden".
+ *
+ * <p>
+ * It manages hidden elements by tracking the number of request made to hide them or show them back. An element will be hidden the first time one asks, and will be showed back only when no one wants it hidden anymore.
+ * </p>
+ */
+var HidingManager = createConstructor(function() {
+    this.attributeName = 'data-hide-requests-count';
+});
+exports.HidingManager = HidingManager;
+var prototype = HidingManager.prototype;
+
+/**
+ * Hides all elements around the given element, by hiding the nodes as close as possible to the root, avoiding any ancestor of the current node.
+ *
+ * @param {HTMLElement} element The element to keep visible
+ *
+ * @return {Function} A function that can revert the operation done by this method.
+ */
+prototype.hideOthers = function(element) {
+    // -------------------------------------------------------------- processing
+
+    var self = this;
+    var reverters = traverseSiblingsUntilRootOfBranch(element, function(element) {
+        self.hide(element);
+        return function () {
+            self.show(element);
+        };
+    });
+
+    reverters = ariaUtilsArray.flatten(reverters);
+    var reverter = ariaUtilsFunction.bind(
+        ariaUtilsArray.forEach,
+        ariaUtilsArray,
+        reverters,
+        ariaUtilsFunction.call
+    );
+
+    // ------------------------------------------------------------------ return
+
+    return reverter;
+};
+
+/**
+ * Show back all elements around the given element, by applying the same traversal as for hideOthers.
+ *
+ * @param {HTMLElement} element The reference element
+ */
+prototype.showOthers = function(element) {
+    var self = this;
+    traverseSiblingsUntilRootOfBranch(element, function(element) {
+        self.show(element);
+    });
+};
+
+
+/**
+ * Registers the request of hiding the given element, and actually hides it if this is the first one.
+ *
+ * <p>
+ * The current implementation uses an attribute "data-hide-requests-count" to store the number of requests.
+ * </p>
+ *
+ * @param {HTMLElement} element The element to hide
+ *
+ * @return {Boolean} Whether or not the element was actually hidden (first request)
+ */
+prototype.hide = function(element) {
+    // ----------------------------------------------------------- destructuring
+
+    var attributeName = this.attributeName;
+
+    var hideRequestsCount = element.getAttribute(attributeName);
+    if (hideRequestsCount == null) {
+        hideRequestsCount = 0;
+    }
+
+    // -------------------------------------------------------------- processing
+
+    var hidden = false;
+
+    if (hideRequestsCount === 0) {
```

Comment:
Under normal circumstances (if no one changes the attribute outside of your `hide` and `show` functions), `hideRequestCount` will never be 0 unless it was null and set to 0 just before... so I think you can simply merge the two `if` blocks together.